src/game.py

```diff
@@ -486,6 +488,13 @@ def getRoadLocations(self, player):
 
         return possible_locations
     
+    def isValidRoadNode(self, node, player):
```

Comment:
Does this fix all road functionality or are there still some bugs?

Reply:
I think this fixes all of it...the only issue I knew of so far was building over others roads. One thing we might want to add a feature for (or we could fix using minimax) is the case where people build roads to others settlements that can never be used and are a waste

Reply:
Whenever you guys want to merge it, looks good to me